test/IRGen/archetype_resilience.sil

```diff
@@ -0,0 +1,27 @@
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) %s -enable-resilience -emit-ir | %FileCheck %s --check-prefix=CHECK
```

Comment:
Can you add this test case to eg, test/IRGen/enum_resilience.swift instead? We already have test/Inputs/resilient_structs.swift which defines some resilient structs, so you don't need to use the mock SDK's Foundation. In fact I'm not sure enable-resilience works with the mock SDK; have you make sure the test fails with your changes backed out?

Reply:
The test fails with my changes backed out. I created a new file / used the SDK because otherwise I can't create a simple test case that reproduces this issue: I need a dynamically sized multi payload enum wherein non of the members are directly defined as opaque archetypes and a there's a class-constarinted T as one of the payloads. taking foundation's URL struct (or UUID or similar) is an easy way to achieve this without adding a lot of code..

Reply:
the resilient_structs module has a bunch of dynamically-sized structs

Reply:
None of witch crash and/or go into the exact same IRGen code path I wanted. You can check it out for yourself by trying any of the structs defined in esilient_struct.swift

Reply:
Ok but what's different about URL vs the resilient_struct structs?

Reply:
The thing is: for most cases, we indirectly worked around this issue because there was either 1) some metadata for the containing enum emitted due to the way collect metadata worked and/or 2) the metadata was found in the cache so we did not try to fetch it / did not hit the assert.

Reply:
If the resilient struct emitted some metadata (which it does in most cases / most record types) the containing enum would also emit metadata itself, which would hide the problem.

Reply:
It's still not clear to me how *from the viewpoint of the client code*, a struct like URL is different from another resilient struct. IRGen should not be looking at the fields of URL; that's the whole point of the struct layout being resilient.

Reply:
created a PR that updates the test case https://github.com/apple/swift/pull/13745